src/tiled/abstractobjecttool.cpp

```diff
@@ -276,6 +288,24 @@ void AbstractObjectTool::showContextMenu(MapObjectItem *clickedObjectItem,
     Utils::setThemeIcon(removeAction, "edit-delete");
     Utils::setThemeIcon(propertiesAction, "document-properties");
 
+    QList<MapObjectItem*> underlyingObjects = listOfObjectItemsAt(scenePos);
+    if (underlyingObjects.size() > 1) {
+        menu.addSeparator();
+
+        QMenu *selectUnderlyingMenu = menu.addMenu(tr("Select an underlying object"));
```

Comment:
What do you think about rather than adding the items here, showing an entirely different menu with just the list of objects, when right-clicking while holding Alt?